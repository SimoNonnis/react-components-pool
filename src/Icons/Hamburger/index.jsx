import React from 'react';
import PropTypes from 'prop-types';

function Hamburger({ width, height, fill, className, classNamePath }) {
  return (
    <svg
      xmlns="http://www.w3.org/2000/svg"
      viewBox="0 0 32 32"
      width={width}
      height={height}
      className={className}
    >
      <title>Hamburger Icon</title>
      <path
        className={classNamePath}
        fill={fill}
        d="M4,10H28a2,2,0,0,0,0-4H4a2,2,0,0,0,0,4Zm24,4H4a2,2,0,0,0,0,4H28a2,2,0,0,0,0-4Zm0,8H4a2,2,0,0,0,0,4H28a2,2,0,0,0,0-4Z"
      />
    </svg>
  );
}

Hamburger.propTypes = {
  width: PropTypes.number,
  height: PropTypes.number,
  fill: PropTypes.string,
  className: PropTypes.string,
  classNamePath: PropTypes.string,
};

Hamburger.defaultProps = {
  width: 32,
  height: 32,
  fill: '#fff',
  className: 'Hamburger',
  classNamePath: 'HamburgerPath',
};

export default Hamburger;
