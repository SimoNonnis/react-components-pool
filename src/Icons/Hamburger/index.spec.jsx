/* eslint-env jest */
import React from 'react';

import Hamburger from './';

describe('Test <Hamburger /> icon', () => {
  it('should render Hamburger icon correctly with default props', () => {
    const wrapper = shallow(<Hamburger />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render Hamburger icon correctly with props', () => {
    const wrapper = shallow(<Hamburger fill="#a4a1a1" className="pof" height={30} width={30} />);
    expect(wrapper).toMatchSnapshot();
  });
});
