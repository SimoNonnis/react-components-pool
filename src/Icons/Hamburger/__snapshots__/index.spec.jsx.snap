// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <Hamburger /> icon should render Hamburger icon correctly with default props 1`] = `
<svg
  className="Hamburger"
  height={32}
  viewBox="0 0 32 32"
  width={32}
  xmlns="http://www.w3.org/2000/svg"
>
  <title>
    Hamburger Icon
  </title>
  <path
    className="HamburgerPath"
    d="M4,10H28a2,2,0,0,0,0-4H4a2,2,0,0,0,0,4Zm24,4H4a2,2,0,0,0,0,4H28a2,2,0,0,0,0-4Zm0,8H4a2,2,0,0,0,0,4H28a2,2,0,0,0,0-4Z"
    fill="#fff"
  />
</svg>
`;

exports[`Test <Hamburger /> icon should render Hamburger icon correctly with props 1`] = `
<svg
  className="pof"
  height={30}
  viewBox="0 0 32 32"
  width={30}
  xmlns="http://www.w3.org/2000/svg"
>
  <title>
    Hamburger Icon
  </title>
  <path
    className="HamburgerPath"
    d="M4,10H28a2,2,0,0,0,0-4H4a2,2,0,0,0,0,4Zm24,4H4a2,2,0,0,0,0,4H28a2,2,0,0,0,0-4Zm0,8H4a2,2,0,0,0,0,4H28a2,2,0,0,0,0-4Z"
    fill="#a4a1a1"
  />
</svg>
`;
